### DISCUSSION TITLE
wayshell only restarts after the second sway reload

### DISCUSSION THREAD

**Discussion by Heus-Sueh** [ldelossa/way-shell#33] [Q&A] [answered]
When reloading the sway the wayshell does not appear, only on the second reload, is this expected?

here is how the wayshell is configured in the sway config:
`exec_always  --no-startup-id  way-shell`

**Comment by ldelossa** [comment]
No need to exec_always, just exec. If you want to reload way-shell often, consider systemd instead.

**Reply by Heus-Sueh** [reply, marked as answer]
> No need to exec_always, just exec. If you want to reload way-shell often, consider systemd instead.

giving just `exec way-shell` makes it not run on reload, just `exec_always way-shell` works, but I found the solution: `exec_always pkill way-shell && way-shell`

my use-case would be this:

https://github.com/ldelossa/way-shell/assets/98482469/5e82bdfa-c007-40c7-9e72-e600eabdf548

**Reply by ldelossa** [reply]
Two instances of way-shell cannot run, and attempting to run a second one will kill the other but also fail to start. 

It is not suggested to restart way-shell like you presented but your work around is fine. 

My suggestion is to use Systemd and the 'sway' target if it suggests for restarting Way-Shell.
